Plugins/MatrixPlugin/MatrixPlugin.cs

```diff
@@ -0,0 +1,136 @@
+using Chummer;
+using Chummer.Plugins;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+using NLog;
+using Microsoft.ApplicationInsights.Channel;
+using System.Diagnostics;
+using System.Xml;
+
+namespace MatrixPlugin
+{
+    [DebuggerDisplay("{" + nameof(GetDebuggerDisplay) + "(),nq}")]
+    public class MatrixPlugin : IPlugin
+    {
+        
+        private static readonly Logger Log = LogManager.GetCurrentClassLogger();
+        List<MatrixAction> Actions;
+        
+        public override string ToString()
+        {
+            return "Matrix Actions Plugin";
+        }
+
+        public void CustomInitialize(frmChummerMain mainControl)
+        {
+            try
```

Comment:
Since you are not doing any UI Stuff, but just initializing you Actions, maybe you wanna spawn a new worker thread here and do it in the background while the main program can continue to load?

Reply:
I am very bad at threading in c#, so i need help in it.
Now i first of all want to fix errors that i already have.